### ISSUE TITLE
Does anyone want http.status_code to report `0` instead of being empty?

### ISSUE THREAD

**Issue by trask** [open-telemetry/opentelemetry-specification#3253]
This could make reporting easier in some backends, but also may set a precedent that is hard to follow for other "conditionally required" attributes, e.g. `http.route`.

**Comment by lmolkova** [comment]
It would be great to have a way to express failures when response was not received. E.g. DNS, connection dropped, timeout. This could be very important for metrics to break down errors by low-cardinality reason.

without an indication of 'no response' status, we're making it harder for users and depends on the backend-specific way to  interpret lack of status code.

**Comment by Oberon00** [comment]
> without an indication of 'no response' status

But the absence of the http.status_code is the indication already?

**Comment by Oberon00** [comment]
>The problem here that if we introduce an attribute later to indicate additional error info, it will be a breaking change for metrics

This being a breaking change sounds like a serious problem. Can't there be a schema entry for new metric attributes so that they can be transparently removed, if this is really a problem?

**Comment by lmolkova** [comment]
> > The problem here that if we introduce an attribute later to indicate additional error info, it will be a breaking change for metrics
> 
> This being a breaking change sounds like a serious problem. Can't there be a schema entry for new metric attributes so that they can be transparently removed, if this is really a problem?

it's a general metrics limitation.  

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md?plain=1#L180-L185

I could not find discussions and reasoning behind it. I believe with the assumption that vendors support the latest and greatest otel version and don't remove attributes by default, time series of before and after attribute was added are no longer comparable and alerts are no longer valid.

**Comment by trask** [comment]
> It would be great to have a way to express failures when response was not received. E.g. DNS, connection dropped, timeout. This could be very important for metrics to break down errors by low-cardinality reason.
>
> without an indication of 'no response' status, we're making it harder for users who'll depend on the backend-specific way to interpret lack of status code.

I believe this is resolved by #205
